crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -413,6 +449,81 @@ impl FormatRule<Suite, PyFormatContext<'_>> for FormatSuite {
             empty_line_after_docstring = false;
         }
 
+        self.between_alternative_blocks_empty_line(statements, &comments, f)?;
+
+        Ok(())
+    }
+}
+
+impl FormatSuite {
+    /// Add an empty line between a function or class and an alternative body.
+    ///
+    /// We only insert an empty if we're between suites in a multi-suite statement. In the
+    /// if-else-statement below, we insert an empty line after the `foo` in the if-block, but none
+    /// after the else-block `foo`, since in the latter case the enclosing suite already adds
+    /// empty lines.
+    ///
+    /// ```python
+    /// if sys.version_info >= (3, 10):
+    ///     def foo():
+    ///         return "new"
+    /// else:
+    ///     def foo():
+    ///         return "old"
+    /// class Bar:
+    ///     pass
+    /// ```
+    fn between_alternative_blocks_empty_line(
+        &self,
+        statements: &Suite,
+        comments: &Comments,
+        f: &mut PyFormatter,
+    ) -> FormatResult<()> {
+        if self.kind.last_suite_in_statement() {
+            // If we're at the end of the current statement, the outer suite will insert one or
+            // two empty lines already.
+            return Ok(());
+        }
+
+        let last_def_or_class = std::iter::successors(
+            statements.last().map(AnyNodeRef::from),
+            AnyNodeRef::last_child_in_body,
+        )
+        // If there is a trailing own line comment, the empty line is inserted before that comment.
+        .take_while(|last_child| !comments.has_trailing_own_line(*last_child))
+        .find(|last_child| {
+            matches!(
+                last_child,
+                AnyNodeRef::StmtFunctionDef(_) | AnyNodeRef::StmtClassDef(_)
+            )
+        });
+        let Some(last_def_or_class) = last_def_or_class else {
+            // An empty line is only inserted for function and class definitions.
+            return Ok(());
+        };
+
```

Comment:
This seems to be a 1:1 copy from

https://github.com/astral-sh/ruff/blob/5f4f4d89710228dfa3908fdaf0de669e9ef81426/crates/ruff_python_formatter/src/statement/suite.rs#L200-L249

It would be great if we can avoid copying the code here